cookbook/bundles/tests.md

```diff
@@ -0,0 +1,255 @@
+
+.. note::
+
+    The steps should be similar with other tests frameworks.
+
+Then the second one is to configure it to use composer's autoloading:
+
+    <?xml version="1.0" encoding="UTF-8"?>
+
+    <!-- http://phpunit.de/manual/4.3/en/appendixes.configuration.html -->
+    <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.3/phpunit.xsd"
+        backupGlobals="false"
+        colors="true"
+        bootstrap="./vendor/autoload.php"
+    >
+        <testsuites>
+            <testsuite name="Test Suite">
+                <directory>./Tests/</directory>
+            </testsuite>
+        </testsuites>
+    </phpunit>
+
+With these two simple steps it becomes possible to run the test suite with the following command:
+
+    vendor/bin/phpunit
+
+Functional CLI tests
+~~~~~~~~~~~~~~~~~~~~
+
+As advised in the official best practices (smoke testing), writing tests for
+commands can be done by simply checking the exit code:
+
+    <?php
+    // File: Tests/Command/DemoCommandTest.php
+
+    namespace Acme\MyBundle\Tests\Command;
+
+    use Acme\MyBundle\Tests\AppKernel;
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+    use Symfony\Component\Console\Tester\ApplicationTester;
+
+    class DemoCommandTest extends \PHPUnit_Framework_TestCase
+    {
+        private $application;
+
+        protected function setUp()
+        {
+            $kernel = new AppKernel('test', false);
+            $application = new Application($kernel);
+            $application->setAutoExit(false);
+            $this->application = new ApplicationTester($application);
+        }
+
+        public function testItRunsSuccessfully()
+        {
+            $exitCode = $this->application->run(array(
+                'command_name' => 'acme:demo',
+                'argument' => 'value',
+                '--option' => 'value',
+            ));
+
+            $this->assertSame(0, $exitCode, $this->application->getDisplay());
+        }
+    }
+
+Functional web tests
+~~~~~~~~~~~~~~~~~~~~
+
+As advised in the official best practices (smoke testing), writing tests for
+controllers can be done by simply checking the status code:
+
+    <?php
+    // File: Tests/Controller/DemoControllerTest.php
+
+    namespace Acme\MyBundle\Tests\Controller;
+
+    use Acme\MyBundle\Tests\AppKernel;
+    use Symfony\Component\HttpFoundation\Request;
+
+    class DemoControllerTest extends \PHPUnit_Framework_TestCase
+    {
+        private $app;
+
+        protected function setUp()
+        {
+            $this->app = new AppKernel('test', false);
+            $this->app->boot();
+        }
+
+        public function testItRunsSuccessfully()
+        {
+            $headers = array('CONTENT_TYPE' => 'application/json');
+            $content = array('parameter' => 'value');
+            $request = Request::create('/demo', 'POST', array(), array(), array(), $headers, $content);
```

Comment:
Each page can have different URI, HTTP method, headers, content and expected status code. Instead of creating a data provider with many parameters that don't really relate to each other, I'd rather create a test method with a meaningful name to describe the behaviour.

Admittedly, the current method name isn't descriptive. I could fix that and maybe add another one? For example a `GET` one?